### PR TITLE
Make SCDF and Skipper compilable for boot 3.x

### PR DESCRIPTION
**Pull request by cppwfs**
* Replaced SocketUtils with TestSocketUtils 
  * It was moved to the test package.
* Migrated httpclient to httpclient5 ** Removed use of httpclient 4.x dependencies added yesterday
* Updated Types in AuditRecord Entity to use JdbcTypeCode
* `DropAuthorizationHeaderRequestRedirectStrategy` requires a bit more scrutiny.  
    Some of the changes required were not exact matches, so had to research.   But there maybe better ways.   

I disabled spring-cloud-common-security-config to push forward .  It was re-added before PR was submitted.

### CODE REVIEW

File: spring-cloud-dataflow-configuration-metadata/pom.xml [hunk 1]
```diff
@@ -49,7 +49,7 @@
 		</dependency>
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-loader</artifactId>
+			<artifactId>spring-boot-loader-classic</artifactId>
```
Comment (onobc) [inline spring-cloud-dataflow-configuration-metadata/pom.xml:52]: What is the significance of this?

File: spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/authorization/S3SignedRedirectRequestServerResource.java [hunk 2]
```diff
@@ -23,7 +23,7 @@
 import org.springframework.boot.builder.SpringApplicationBuilder;
 import org.springframework.cloud.dataflow.container.registry.authorization.support.S3SignedRedirectRequestServerApplication;
 import org.springframework.context.ConfigurableApplicationContext;
-import org.springframework.util.SocketUtils;
+import org.springframework.test.util.TestSocketUtils;
```
Comment (onobc) [inline spring-cloud-dataflow-configuration-metadata/src/test/java/org/springframework/cloud/dataflow/container/registry/authorization/S3SignedRedirectRequestServerResource.java:26]: 🎉 finally people are using it! 😉

File: spring-cloud-dataflow-container-registry/src/main/java/org/springframework/cloud/dataflow/container/registry/ContainerImageRestTemplateFactory.java [hunk 3]
```diff
@@ -171,19 +177,25 @@ public void checkServerTrusted(java.security.cert.X509Certificate[] certs, Strin
...
 		// Install trust manager to SSL Context.
 		sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+		ConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE);
+
+		Registry<ConnectionSocketFactory> socketFactoryRegistry =
+			RegistryBuilder.<ConnectionSocketFactory> create()
+				.register("https", sslsf)
+				.register("http", new PlainConnectionSocketFactory())
+				.build();
+		final BasicHttpClientConnectionManager connectionManager = new BasicHttpClientConnectionManager(socketFactoryRegistry);
```
Comment (onobc) [inline spring-cloud-dataflow-container-registry/src/main/java/org/springframework/cloud/dataflow/container/registry/ContainerImageRestTemplateFactory.java:187]: Please remove `final`.

File: spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/AuditRecord.java [hunk 4]
```diff
@@ -58,7 +59,7 @@ public class AuditRecord {
 	private String correlationId;
 
 	@Lob
-	@Type(type = "org.hibernate.type.TextType")
+	@JdbcTypeCode(Types.LONGVARCHAR)
```
Comment (onobc) [inline spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/AuditRecord.java:62]: I know we had a headache around LOB support specifically w/ Postgres. Hopefully the DB ITs will and ITs will ferret this out if it is a problem. The pain w/ the LOB support was that not only was it different across vendors, but also versions of a vendor.

File: spring-cloud-dataflow-container-registry/src/main/java/org/springframework/cloud/dataflow/container/registry/ContainerImageRestTemplateFactory.java [hunk 5]
```diff
@@ -171,19 +177,25 @@ public void checkServerTrusted(java.security.cert.X509Certificate[] certs, Strin
 		SSLContext sslContext = SSLContext.getInstance("SSL");
 		// Install trust manager to SSL Context.
 		sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+		ConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE);
```
Comment (onobc) [inline spring-cloud-dataflow-container-registry/src/main/java/org/springframework/cloud/dataflow/container/registry/ContainerImageRestTemplateFactory.java:180]: It took me a bit to figure out what was new here. My final analysis is that we just have to jump through another hoop to construct the required http client builder. 

I reworked this a bit to abstract out the "get http client builder" which hides the new extra unimportant confusing steps.  Wdyt about replacing this chunk w/ the following?:

```suggestion
		// Create a RestTemplate that uses custom request factory
		return initRestTemplate(
				httpClientBuilder(sslContext),
				withHttpProxy,
				extra);
	}

	private HttpClientBuilder httpClientBuilder(SSLContext sslContext) {
		// Register http/s connection factories
		Lookup<ConnectionSocketFactory> connSocketFactoryLookup = RegistryBuilder.<ConnectionSocketFactory> create()
				.register("https", new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE))
				.register("http", new PlainConnectionSocketFactory())
				.build();
		return HttpClients.custom()
				.setConnectionManager(new BasicHttpClientConnectionManager(connSocketFactoryLookup));
	}
```

**Review by onobc — CHANGES_REQUESTED** [review]
Nice work @cppwfs ! I left a couple of suggestions, but other than that - LGTM.

File: spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/AuditRecord.java (hunk 4 above)
Comment (corneil) [reply spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/AuditRecord.java:62]: Running Postgres Smoke Tests locally will quickly identify if this is a valid change.

File: spring-cloud-dataflow-configuration-metadata/pom.xml (hunk 1 above)
Comment (cppwfs) [reply spring-cloud-dataflow-configuration-metadata/pom.xml:52]: `org.springframework.boot.loader.archive.JarFileArchive` (The one dataflow uses) moved to this dependency.   There is another `org.springframework.boot.loader.launch.JarFileArchive`.  I chose the one that dataflow has used in the past.

**Review by onobc — APPROVED** [review]
LGTM - thanks for the updates @cppwfs

File: spring-cloud-skipper/spring-cloud-skipper-client/src/main/java/org/springframework/cloud/skipper/client/util/HttpClientConfigurer.java [hunk 6]
```diff
@@ -75,12 +84,22 @@ public HttpClientConfigurer basicAuthCredentials(String username, String passwor
...
-
+		Lookup<ConnectionSocketFactory> connSocketFactoryLookup = RegistryBuilder.<ConnectionSocketFactory> create()
+			.register("https", new SSLConnectionSocketFactory(HttpUtils.buildCertificateIgnoringSslContext(), NoopHostnameVerifier.INSTANCE))
+			.register("http", new PlainConnectionSocketFactory())
+			.build();
+		httpClientBuilder.setConnectionManager(new BasicHttpClientConnectionManager(connSocketFactoryLookup));
 		return this;
 	}
-
+	private HttpClientBuilder httpClientBuilder(SSLContext sslContext) {
```
Comment (onobc) [inline spring-cloud-skipper/spring-cloud-skipper-client/src/main/java/org/springframework/cloud/skipper/client/util/HttpClientConfigurer.java:94]: I don't think this is currently being used in this class.

**Review by onobc — APPROVED** [review]
@cppwfs LGTM - one minor suggestion. Other than that 🍏

File: spring-cloud-skipper/spring-cloud-skipper-client/src/main/java/org/springframework/cloud/skipper/client/util/HttpClientConfigurer.java (hunk 6 above)
Comment (cppwfs) [reply spring-cloud-skipper/spring-cloud-skipper-client/src/main/java/org/springframework/cloud/skipper/client/util/HttpClientConfigurer.java:94]: Good catch.  I was going to extract the code from a method and make it re-usable.  Then realized that it was only being used there.   Then forgot to remove the new method.

**Comment by cppwfs** [comment]
Squashed and merged into main-3

**Pull request closed without merging**